client/src/BidList.js

```diff
@@ -33,10 +34,27 @@ const BuyerTenderList = () => {
 		}
 	};
 
+	const fetchTender = async (tenderId) => {
+		setLoading(true);
+		try {
+			const data = await get(`/api/tenders/${tenderId}`);
+			setTender(data.resource);
+			setTenderErrorMsg(null);
+		} catch (error) {
+			setTenderErrorMsg("Error fetching tenders!");
+		} finally {
+			setLoading(false);
+		}
+	};
+
 	useEffect(() => {
 		fetchBids(tenderId, currentPage);
 	}, [currentPage, tenderId]);
 
+	useEffect(() => {
+		fetchTender(tenderId);
+	}, [tenderId]);
+
```

Comment:
Why do you have `useEffect` twice?

Reply:
thats because one is fetching bids and current page and the other one is fetching tenders. and they are using different endpoints the functions

Reply:
That's wrong. There should only be one useEffect

Reply:
will change it to use one use effect